clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java

```diff
@@ -99,7 +100,7 @@ public String name() {
     /**
      * Return the value or null. Null is returned if the config is unset or if isSensitive is true.
      */
-    public String value() {
+    public Optional<String> value() {
```

Comment:
Note that `ConfigEntry` is considered a public API. This change is not backwards compatible, but we could consider exposing a separate method to expose the value as an optional. There is a process for changing public APIs: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals.